src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -481,5 +489,80 @@ private static string GetTempAssemblyName(AssemblyName parent, string ns)
         {
             return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + ns.GetHashCode());
         }
+
+        private static void ParseReferences(string value, Dictionary<string, string> dictionary)
+        {
```

Comment:
Can this method be called once in the Run method and the resulting dictionary stored in a static dictionary?

Reply:
We don't need call this method if there's no reference parameter.

Reply:
So then do:
```csharp
if (!string.IsNullOrEmpty(s_references))
{
    ParseReferences();
}
```
You don't need to pass s_references as it's a static method which can access static properties. Store it in a static field called s_referenceDictionary. That way it's only called if references are passed, but it's only called once. Right now you call it over and over again.